crates/ironrdp-displaycontrol/src/client.rs

```diff
@@ -1,14 +1,47 @@
 use crate::{
-    pdu::{DisplayControlMonitorLayout, DisplayControlPdu, MonitorLayoutEntry},
+    pdu::{DisplayControlCapabilities, DisplayControlPdu},
     CHANNEL_NAME,
 };
 use ironrdp_dvc::{encode_dvc_messages, DvcClientProcessor, DvcMessage, DvcProcessor};
-use ironrdp_pdu::PduResult;
+use ironrdp_pdu::{cursor::ReadCursor, PduDecode, PduResult};
 use ironrdp_svc::{impl_as_any, ChannelFlags, SvcMessage};
 use tracing::debug;
 
 /// A client for the Display Control Virtual Channel.
-pub struct DisplayControlClient {}
+pub struct DisplayControlClient {
+    /// A callback that will be called when capabilities are received from the server.
+    /// If no callback is set, a default (inert) callback will be used.
+    on_capabilities_received: OnCapabilitiesReceived,
+    /// Indicates whether the capabilities have been received from the server.
+    ready: bool,
+}
+
+impl DisplayControlClient {
+    /// Creates a new [`DisplayControlClient`] with the given `callback`.
+    ///
+    /// The `callback` will be called when capabilities are received from the server.
+    /// It is important to note that the channel will not be fully operational until the capabilities are received.
+    /// Attempting to send messages before the capabilities are received will result in an error or a silent failure.
+    pub fn new<F>(callback: F) -> Self
+    where
+        F: Fn(DisplayControlCapabilities) -> PduResult<Vec<DvcMessage>> + Send + 'static,
+    {
+        Self {
+            on_capabilities_received: Box::new(callback),
+            ready: false,
+        }
+    }
+
+    pub fn ready(&self) -> bool {
```

Comment:
`(optional)`

Idea: To enforce correct API usage, we could return a token here, which  is required to call `encode_monitor message` later:

```rust
#[non_exhaustive]
#[derive(Clone, Copy, Debug)]
pub struct DisplayControlReadyToken;

    pub fn poll_ready(&self) -> Option<DisplayControlReadyToken> {
        ready.then(|| DisplayControlReadyToken)
    }

    pub fn encode_monitor(&self, _token: DisplayControlReadyToken, channel_id: u32, width: u32, height: u32) -> PduResult<Vec<SvcMessage>> {
        // ...
    }
}
```

Reply:
More trouble than it's worth, IMO